### PR TITLE
Remove merge syft sbom python script

### PR DESCRIPTION
**Pull request by mkosiarc**
Removing the script from this repo and migrating it to build-tasks-dockerfiles, to a separate image for sbom utility scripts.

See https://github.com/konflux-ci/build-tasks-dockerfiles/pull/154

# Maintainers will complete the following section

- [ ] Commit messages are descriptive enough
- [ ] Code coverage from testing does not decrease and new code is covered
- [ ] Docs updated (if applicable)
- [ ] Docs links in the code are still valid (if docs were updated)

**Note:** if the contribution is external (not from an organization member), the CI
pipeline will not run automatically. After verifying that the CI is safe to run:

- [approve GitHub Actions workflows][approve-gh-actions] by clicking a button
- approve the Red Hat Trusted App Pipeline container build by commenting `/ok-to-test`
  (as is the [standard for Pipelines as Code][pac-running-pipeline])

[approve-gh-actions]: https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks
[pac-running-pipeline]: https://pipelinesascode.com/docs/guide/running/#running-the-pipelinerun

### CODE REVIEW

File: tests/unit/data/sboms/cachi2.bom.json
[the remark this answers is not in the document]
Comment (mkosiarc) [reply tests/unit/data/sboms/cachi2.bom.json:1]: not a problem, updated to keep the file

File: Dockerfile [hunk 1]
```diff
@@ -46,7 +46,6 @@ COPY --from=golang_121 /usr/local/go /usr/local/go/go1.21
 COPY --from=node_223 /usr/local/lib/node_modules/corepack /usr/local/lib/corepack
 COPY --from=node_223 /usr/local/bin/node /usr/local/bin/node
 COPY --from=builder /venv /venv
-COPY --from=builder /src/utils/merge_syft_sbom.py /usr/local/bin/merge_syft_sbom
```
Comment (eskultety) [inline Dockerfile:49]: **Commit message:**

Nitpicking about the commit message, let's improve the wording just a bit. How about:
_Removing the script from this repo and migrating it to a place and organization more appropriate to host this scrip than this project, see https://github.com/konflux-ci/build-tasks-dockerfiles/pull/154_

Comment (mkosiarc) [reply Dockerfile:49]: Erik, please :D

File: tests/unit/data/sboms/cachi2.bom.json
[the remark this answers is not in the document]
Comment (a-ovchinnikov) [reply tests/unit/data/sboms/cachi2.bom.json:1]: Thank you!

File: tests/unit/data/sboms/syft.bom.json
Comment (a-ovchinnikov) [inline tests/unit/data/sboms/syft.bom.json:1]: I have just realized that this one is needed as well (as an example of a non-Cachi2 generated SBOM). Could you please keep it too?

Comment (mkosiarc) [reply tests/unit/data/sboms/syft.bom.json:1]: Restored the file as well.

Comment (a-ovchinnikov) [reply tests/unit/data/sboms/syft.bom.json:1]: Thank you!

**Review by a-ovchinnikov — APPROVED** [review]
LGTM

**Review by slimreaper35 — APPROVED** [review]

**Pull request merged**